Subject: fix tg3 flow ctrl
From: Michael Chan <mchan@broadcom.com>                                                                                                                                
References: 390314

If we are trying to establish link in parallel detect mode,
the flow control settings may not match.
If we do not enter the if statement to do nothing, we will keep
autonegotiating forever and never establish link. 

Signed-off-by: Olaf Hering <olh@suse.de>

---
 drivers/net/tg3.c |  107 +++++++++++++++++++++---------------------------------
 1 file changed, 43 insertions(+), 64 deletions(-)

--- a/drivers/net/tg3.c
+++ b/drivers/net/tg3.c
@@ -3168,8 +3168,7 @@ static int tg3_setup_fiber_mii_phy(struc
 	err |= tg3_readphy(tp, MII_BMCR, &bmcr);
 
 	if ((tp->link_config.autoneg == AUTONEG_ENABLE) && !force_reset &&
-	    (tp->tg3_flags2 & TG3_FLG2_PARALLEL_DETECT) &&
-	     tp->link_config.flowctrl == tp->link_config.active_flowctrl) {
+           (tp->tg3_flags2 & TG3_FLG2_PARALLEL_DETECT)) {
 		/* do nothing, just check for link up at the end */
 	} else if (tp->link_config.autoneg == AUTONEG_ENABLE) {
 		u32 adv, new_adv;
